src/content/docs/core-concepts/query-pipeline.mdx

```diff
@@ -0,0 +1,102 @@
+---
+title: Query Pipeline
+description: A step-by-step introduction on how to use swiftide for generating answers with a query pipeline.
+sidebar:
+  order: 0
+---
+
+After indexing data, you can use a query pipeline to answer questions, using the indexed data, together with various
+transformations, to generate the most relevant context for the LLM.
+
+Query pipelines, similar to indexing pipelines, function as parallel streams.
+
+In the pipeline, a `Query<states::Pending>` starts its journey of going through
+a series of transformations, before being used to generate an answer.
+
+By default, a similarity search is used on a single embedding.
+
+import { Steps } from "@astrojs/starlight/components";
+
+### A query pipeline pipeline step-by-step
+
+<Steps>
+
+1. Start with the default query pipeline that does a similarity search with a single embedding:
+
+   ```rust
+   let pipeline = query::Pipeline::default();
+   ```
+
+2. Then generate subquestions, improving the semantic coverage of the query:
+
+   ```rust
+   pipeline.then_transform_query(query_transformers::GenerateSubQuestions::from_client(openai_client.clone()))
+   ```
+
+   This transformer takes the **current** query, generates subquestions and sets that as the new query.
+
+   **A query transformer takes a `Query<states::Pending>` and implements the `TransformQuery` trait. Query transformers always return a `Result<Query<states::Pending>>`. For each state, `Query` provides an interface for state transitioning.**
+
+3. Closures are also supported at every step:
+
+   ```rust
+   pipeline.then_transform_query(|mut query| {
+     query.transform_query("Hello world")
+
+     Ok(query)
+   });
+   ```
+
+4. Now we need to generate an embedding for the transformed query:
+
+   ```rust
+   pipeline.transform_query(query_transformers::Embed::from_client(openai_client.clone()))
+   ```
+
+   An embedding is added to the query, used for retrieval. Note that multiple embeddings are not (yet!) supported.
```

Comment:
So the result of the previous step, the generated subquestions, are they just concatenated to the original query and then embedded as one big string?

Reply:
It takes whatever is in current, if the transformer keeps the original as well it will. Good point, I think the intro should lead with how that works.